lib/models/user_data.dart

```diff
@@ -28,3 +29,38 @@ abstract class UserData extends ChangeNotifier {
     ''';
   }
 }
+
+/// Extension [TimeCheck] on [UserData].
+///
+/// This extension provides additional methods related to time checking for [UserData] objects.
+extension TimeCheck on UserData {
```

Comment:
I would personally separate this to another file. On iOS platfrom, we have an convention for naming extension files as `Root+ExtensionName`, in this case it would be `UserData+TimeCheck`. 
Let me know your opinion :)

Reply:
Yes, thank you, you are right. I moved it to `models/extensions/user_data_time_check_extensions.dart`. In Dart/Flutter it is customary to use snake_case. Is that ok?

Reply:
I ended up making it a class feature of `Canteen`. See Alex's comment.